### PR TITLE
PLT-1278 Updated the basic app tutorial in the ReadTheDocs.

### PR DESCRIPTION
**Pull request by koslambrou**
* Fixed some outdated information

* Changed the instructions to be able to run the contract in the emulator instead of the playground

* Small refactoring on the Plutus emulator API

Pre-submit checklist:
- Branch
    - [ ] Tests are provided (if possible)
    - [x] Commit sequence broadly makes sense
    - [x] Key commits have useful messages
    - [x] Formatting, PNG optimization, etc. are updated
    - [x] Important changes are reflected in changelog.d of the affected packages
- PR
    - [x] Self-reviewed the diff
    - [x] Useful pull request description
    - [ ] Reference the ADR in the PR and reference the PR in the ADR (if revelant)
    - [x] Reviewer requested

### CODE REVIEW

File: doc/plutus/tutorials/basic-apps.rst [hunk 1]
```diff
@@ -87,15 +87,15 @@ In Plutus apps, the mechanism for requesting inputs is called :term:`endpoints <
...
+You can build a schema using the :hsobj:`Plutus.Contract.Endpoint` type family to construct individual endpoint types, and the ``.\/`` operator to combine them.
 
 .. literalinclude:: BasicApps.hs
    :start-after: BLOCK4
    :end-before: BLOCK5
 
 The ``SplitSchema`` defines two endpoints, ``lock`` and ``unlock``.
 Each endpoint declaration contains the endpoint's name and its type.
-Note that ``SplitData`` has two ``PubKeyHash`` fields for the recipients, whereas the endpoint has two ``Wallet`` fields (the wallet type is used in the Playground as the identity of a simulated agent).
+Note that ``SplitData`` has two :hsobj:`Plutus.V1.Ledger.Api.PubKeyHash` fields for the recipients, whereas the endpoint has two ``Wallet`` fields (the wallet type is used in the Plutus apps emulator as the identity of a simulated agent).
```
Comment (sjoerdvisscher) [inline doc/plutus/tutorials/basic-apps.rst:98]: The `PubKeyHash`es and `Wallet`s are both `Address`es now.

File: doc/plutus/tutorials/basic-apps.rst [hunk 2]
```diff
@@ -148,51 +141,195 @@ All that's missing now is the code for retrieving the funds, and some glue to pu
...
 The following action sequence results in two transactions that lock the funds and then distribute them to the two recipients.
 
-.. figure:: images/playground-basic-app-simulation.png
+.. literalinclude:: BasicApps.hs
+   :start-after: BLOCK11
+   :end-before: BLOCK12
+
+You should see an output similar to what follows::
+
+  [INFO] Slot 0: TxnValidate 43ba666cc8a22a04b63a3b605ce14146dfa5ed999986625ad90c1bc16dabdd84 [  ]
```
Comment (sjoerdvisscher) [inline doc/plutus/tutorials/basic-apps.rst:181]: Make this a golden test output and include the file here? Not sure how much trouble that is.

**Review by sjoerdvisscher — APPROVED** [review]
I think this is a much more useful way of testing contracts than the playground, nice!

File: doc/plutus/tutorials/basic-apps.rst (hunk 2 above)
Comment (koslambrou) [reply doc/plutus/tutorials/basic-apps.rst]: Good idea!

Done. Was fairly quick.

**Review by joseph-fajen — APPROVED** [review]
Should any of these references be pointing to Plutus.V2 counterparts for any reason? I don't know if this applies here in this context. 

- Plutus.V1.Ledger.Api
- Plutus.V1.Ledger.Api.Address
- Plutus.V1.Ledger.Api.ScriptContext
- Plutus.V1.Ledger.Value

**Comment by koslambrou** [comment]
@joseph-fajen Good point. For this tutorial however, we're staying with PlutusV1. I'll create a story to create an updated version it.

**Comment by koslambrou** [comment]
@zeme-iohk Testing locally, that seems to only occur when running `nix-build -A docs.site`. Tried adding the file with the `data-files` cabal field like so, but doesn't change anything:
```
  data-files:
    plutus/tutorials/basic-apps-trace.txt
```

**Comment by koslambrou** [comment]
Ok, found the issue. Had to change the `default.nix` to include text files.

**Pull request merged**